registry/handlers/app.go

```diff
@@ -546,8 +546,16 @@ func (app *App) configureRedis(configuration *configuration.Configuration) {
 			}
 
 			// authorize the connection
+			authArgs := make([]interface{}, 0, 2)
+			if configuration.Redis.Username != "" {
+				authArgs = append(authArgs, configuration.Redis.Username)
+			}
 			if configuration.Redis.Password != "" {
-				if _, err = conn.Do("AUTH", configuration.Redis.Password); err != nil {
+				authArgs = append(authArgs, configuration.Redis.Password)
+			}
+
+			if len(authArgs) > 0 {
```

Comment:
I left a comment earlier, but I'll repeat it here: what happens if you are forced to run a lower version than `v6`? The previous code only required `Password`, this PR requires both `Username` and `Password` and if either of them is missing there is no `AUTH` performed.

Reply:
The ACL feature in redis is backward compatible, the user can use command `AUTH [username] password` to perform the authorization, the username is optional, so the PR not bring the breaking changes, only attach the username when user specifies it, otherwise it is same with the previous behavior.